casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -243,25 +253,32 @@ object BlockReceiver {
           val markReceivedAndStore =
             for {
               // Save block to block store, resolve parents to request
-              _ <- BlockStore[F].put(block)
+              blockStored <- BlockStore[F].contains(block.blockHash)
+              _ <- (BlockStore[F].put(block) *> BlockRetriever[F].ackReceived(block.blockHash))
+                    .whenA(!blockStored)
```

Comment:
Why calling of block retriever is skipped if block is the store. Block retriever should know if block is received even if in the store to remove it from internal state. Did you see issues with that?

Reply:
Do you mean `BlockRetriever` doesn't restore state from disk after node starts unlike `BlockStore` so it can request for blocks even if it already received and stored?